diff --git a/App.js b/App.js
index c1ce732..f439407 100644
--- a/App.js
+++ b/App.js
@@ -16,7 +16,7 @@ const theme = {
 export default class App extends Component {
   render() {
     return (
-        <StoreProvider store={store}>
+         <StoreProvider store={store}>
           <PaperProvider theme={theme}>
             <AppContainer />
           </PaperProvider>
diff --git a/app/actions/types.actions.js b/app/actions/types.actions.js
index 4c6a4ec..a3343ae 100644
--- a/app/actions/types.actions.js
+++ b/app/actions/types.actions.js
@@ -3,4 +3,21 @@ The purpose of this file to have all the action types as constant so that there
 Note: This is a action types file.
 Please remove it when developing a new app.
 */
-export const ADD_TEST = 'ADD_TEST';
\ No newline at end of file
+export const FETCH_PRODUCTS = 'FETCH_PRODUCTS';
+export const ADD_TO_CART = 'ADD_TO_CART';
+export const REMOVE_FROM_CART = 'REMOVE_FROM_CART';
+export const ADD_ORDER = 'ADD_ORDER';
+export const EMPTY_CART = 'EMPTY_CART';
+export const USER_PROFILE = 'SAVE_USER';
+
+import axiosObj from '../utils/api.helper';
+
+export const getUser = user => ({
+    type: 'GET_USER',
+    payload: axiosObj.post('/user/login', user),
+});
+
+export const getToken = token => ({
+    type: 'GET_TOKEN',
+    payload: token,
+});
\ No newline at end of file
diff --git a/app/containers/consumerTabContainer.js b/app/containers/consumerTabContainer.js
index 51d922c..3b47ddb 100644
--- a/app/containers/consumerTabContainer.js
+++ b/app/containers/consumerTabContainer.js
@@ -57,7 +57,11 @@ export default class ConsumerTabs extends Component {
                   name="shopping"
                 />
                 <Text
-                  style={{ textAlignVertical: "center", textAlign: "center" }}
+                  style={{
+                    textAlignVertical: "center",
+                    textAlign: "center",
+                    marginLeft: 5
+                  }}
                 >
                   Feed
                 </Text>
@@ -74,11 +78,19 @@ export default class ConsumerTabs extends Component {
                   type="MaterialCommunityIcons"
                   name="cart"
                 />
-                <Text>Cart</Text>
+                <Text
+                  style={{
+                    textAlignVertical: "center",
+                    textAlign: "center",
+                    marginLeft: 5
+                  }}
+                >
+                  Cart
+                </Text>
               </TabHeading>
             }
           >
-          <ConsumerC />  
+            <ConsumerC />
           </Tab>
           <Tab
             heading={
@@ -88,7 +100,15 @@ export default class ConsumerTabs extends Component {
                   type="MaterialIcons"
                   name="local-shipping"
                 />
-                <Text>Orders</Text>
+                <Text
+                  style={{
+                    textAlignVertical: "center",
+                    textAlign: "center",
+                    marginLeft: 5
+                  }}
+                >
+                  Orders
+                </Text>
               </TabHeading>
             }
           >
diff --git a/app/containers/customerCont/customerCart.js b/app/containers/customerCont/customerCart.js
index 9d75975..52eb775 100644
--- a/app/containers/customerCont/customerCart.js
+++ b/app/containers/customerCont/customerCart.js
@@ -133,6 +133,9 @@ export default class ConsumerC extends Component {
                 <Card.Actions>
                   <TextInput
                     style={{ width: 213, height: 56, marginRight: 60 }}
+                    selectionColor="#757575"
+                    underlineColor="#757575"
+                    
                     mode="outlined"
                     textAlignVertical="center"
                     label="Quantity"
diff --git a/app/reducers/root.reducer.js b/app/reducers/root.reducer.js
index 7f4cc90..3cf334b 100644
--- a/app/reducers/root.reducer.js
+++ b/app/reducers/root.reducer.js
@@ -1,6 +1,8 @@
-import {combineReducers} from 'redux';
-import sample from './sample.reducer';
+import { combineReducers } from 'redux';
+import user from './userReducer';
+import userToken from './tokenReducer'
 
 export default combineReducers({
-    sample
-})
\ No newline at end of file
+  user,
+  userToken,
+});
diff --git a/app/reducers/sample.reducer.js b/app/reducers/sample.reducer.js
deleted file mode 100644
index aa29d50..0000000
--- a/app/reducers/sample.reducer.js
+++ /dev/null
@@ -1,16 +0,0 @@
-/*
-Note: This is a sample reducer.
-Please remove it when developing a new app.
-*/
-
-export default sample = (state = [], action) => {
-    switch (action.type){
-        case 'ADD_TODO':
-            return [ ...state, {
-                id: action.id,
-                text: action.text,
-            }];
-        default:
-            return state;
-    }
-}
\ No newline at end of file
diff --git a/app/store.js b/app/store.js
index a57cca8..cf13b7e 100644
--- a/app/store.js
+++ b/app/store.js
@@ -1,4 +1,7 @@
-import {createStore} from "redux";
+import {createStore, compose} from "redux";
+
+// import {autoRehydrate} from 'redux-persist';
+
 import rootReducer from './reducers/root.reducer';
 
 export default createStore(rootReducer);
\ No newline at end of file
diff --git a/app/utils/api.helper.js b/app/utils/api.helper.js
index f2f8ca9..d18ad49 100644
--- a/app/utils/api.helper.js
+++ b/app/utils/api.helper.js
@@ -7,12 +7,12 @@ const baseUrl = 'https://hc-india.hotwax.co';
 //     timeout: 10000
 // });
 
-let axiosObj = axios.create({
+let axiosObject = axios.create({
     baseURL: "https://kissan-haat.herokuapp.com/api",
     timeout: 10000
 });
 
-// axiosObject.interceptors.request.use(async (config) => {
+// axiosObjectect.interceptors.request.use(async (config) => {
 //     const token = await AsyncStorage.getItem("authToken");
 //     if (token != null) {
 //         config.headers.Authorization = 'Bearer ' + token;
@@ -22,7 +22,7 @@ let axiosObj = axios.create({
 //     return Promise.reject(error);
 // });
 
-// axiosObject.interceptors.response.use(function (config) {
+// axiosObjectect.interceptors.response.use(function (config) {
 //     return config;
 // }, function (error) {
 //     return Promise.reject(error);
@@ -30,7 +30,7 @@ let axiosObj = axios.create({
 
 const apiHelper = {
     login: (user) => {
-        return axiosObj.post('/user/login', user);
+        return axiosObject.post('/user/login', user);
     },
     logout: () => {
         return AsyncStorage.clear()
@@ -39,8 +39,9 @@ const apiHelper = {
         return axiosObject.post('/user-profile', {});
     },
     getCropList: () => {
-        return axiosObj.get('/cropList');
+        return axiosObject.get('/cropList');
     }
 }
 
-export default apiHelper;
\ No newline at end of file
+export default apiHelper;
+export  const axiosObj = axiosObject
\ No newline at end of file
diff --git a/package.json b/package.json
index f37c071..ba9ff7e 100644
--- a/package.json
+++ b/package.json
@@ -21,7 +21,9 @@
     "react-native-vector-icons": "^6.6.0",
     "react-navigation": "^3.5.1",
     "react-redux": "^6.0.1",
-    "redux": "^4.0.1"
+    "redux": "^4.0.4",
+    "redux-persist": "^6.0.0",
+    "redux-thunk": "^2.3.0"
   },
   "devDependencies": {
     "babel-core": "^7.0.0-bridge.0",
